yarn-project/noir-protocol-circuits/src/crates/types/src/address.nr

```diff
@@ -128,3 +132,36 @@ impl EthAddress{
     }
 }
 
+// Partial address
```

Comment:
I like that it is not just a random field anymore. But is is still a really confusing value 😬. The name really make it seem like something that you could be using as an address somewhere, but it is really a code identifier. I propose that we should alter the name to be `CodeIdentifier` or something along those lines 🤷.

Reply:
There is also contract address salt which is unrelated to code so I think `CodeIdentifier` is not a good name 🤣